biogeochem/EDPatchDynamicsMod.F90

```diff
@@ -1019,411 +1127,697 @@ subroutine set_patchno( currentSite )
+             do dcmpy=1,ndcmpy
+                 dcmpy_frac = GetDecompyFrac(pft,dcmpy)
+                 do sl = 1,currentSite%nlevsoil
+                     donatable_mass = num_dead_trees * (fnrt_m+store_m) * currentSite%rootfrac_scr(sl)
+                     new_litt%root_fines(dcmpy,sl) = new_litt%root_fines(dcmpy,sl) + &
+                                                     donatable_mass*donate_m2*dcmpy_frac
+                     curr_litt%root_fines(dcmpy,sl) = curr_litt%root_fines(dcmpy,sl) + &
+                                                      donatable_mass*retain_m2*dcmpy_frac
+                 end do
+             end do
+
+             ! Track as diagnostic fluxes
+             flux_diags%leaf_litter_input(pft) = &
+                  flux_diags%leaf_litter_input(pft) + &
+                  num_dead_trees * (leaf_m+repro_m) * (1.0_r8-currentCohort%fraction_crown_burned)
+
+             flux_diags%root_litter_input(pft) = &
+                  flux_diags%root_litter_input(pft) + &
+                  (fnrt_m + store_m) * num_dead_trees
+             
+             ! coarse root biomass per tree
+             bcroot = (sapw_m + struct_m) * (1.0_r8 - EDPftvarcon_inst%allom_agb_frac(pft) )
       
              ! below ground coarse woody debris from burned trees
              do c = 1,ncwd
-                new_patch%cwd_bg(c)    = new_patch%cwd_bg(c) + dead_tree_density * SF_val_CWD_frac(c) * bcroot
-                currentPatch%cwd_bg(c) = currentPatch%cwd_bg(c) + dead_tree_density * SF_val_CWD_frac(c) * bcroot
+                do sl = 1,currentSite%nlevsoil
+                   donatable_mass =  num_dead_trees * SF_val_CWD_frac(c) * &
+                         bcroot * currentSite%rootfrac_scr(sl)
+
+                   new_litt%bg_cwd(c,sl) = new_litt%bg_cwd(c,sl) + &
+                         donatable_mass * donate_m2
+                   curr_litt%bg_cwd(c,sl) = curr_litt%bg_cwd(c,sl) + &
+                         donatable_mass * retain_m2
+
+                   ! track diagnostics
+                   flux_diags%cwd_bg_input(c) = &
+                         flux_diags%cwd_bg_input(c) + &
+                         donatable_mass
+                enddo
+             end do
 
-                ! track as diagnostic fluxes
-                currentSite%CWD_BG_diagnostic_input_carbonflux(c) = currentSite%CWD_BG_diagnostic_input_carbonflux(c) + &
-                     SF_val_CWD_frac(c) * bcroot * currentCohort%fire_mort * currentCohort%n * &
-                     hlm_days_per_year / AREA
-             enddo
+             ! stem biomass per tree
+             bstem  = (sapw_m + struct_m) * EDPftvarcon_inst%allom_agb_frac(pft)
 
-             ! above ground coarse woody debris from unburned twigs and small branches
+             ! Above ground coarse woody debris from twigs and small branches
+             ! a portion of this pool may burn
              do c = 1,2
```

Comment:
Suggestion to reduce repeated lines of code:
```suggestion
             do c = 1,ncwd
                 donatable_mass = num_dead_trees * SF_val_CWD_frac(c) * bstem
                 if (c == 1 .or. c == 2) then
                      donatable_mass = donatable_mass * (1.0_r8-currentCohort%fraction_crown_burned)
                      burned_mass = num_dead_trees * SF_val_CWD_frac(c) * bstem * & 
                      currentCohort%fraction_crown_burned
                      site_mass%burn_flux_to_atm = site_mass%burn_flux_to_atm + burned_mass
                endif
                new_litt%ag_cwd(c) = new_litt%ag_cwd(c) + donatable_mass * donate_m2
                curr_litt%ag_cwd(c) = curr_litt%ag_cwd(c) + donatable_mass * retain_m2
                flux_diags%cwd_ag_input(c) = flux_diags%cwd_ag_input(c) + donatable_mass
             enddo   
                  
```

Reply:
I like this.

It's hard on this thread, to tell what exactly would be removed if I commit the suggestion.  Does this include the removal of the loop here (L1582):

https://github.com/NGEET/fates/pull/524/files/3d56ddd8b5c00ac4ee98db7aedf9fa34f1097d74#diff-531e45e75218bb11634691970470c90bR1582

and it's contents?  I think that needs to be removed too.  I suppose i could just commit and see...

Reply:
Yeah, sorry.  First time suggesting more than a one line change.  Yup, `the n = 3,4` section would be removed.

Reply:
great, will incorporate your changes soon